src/modules/daml.rs

```diff
@@ -0,0 +1,120 @@
+use super::{Context, Module, ModuleConfig};
+use crate::configs::daml::DamlConfig;
+use crate::formatter::{StringFormatter, VersionFormatter};
+
+const DAML_SDK_VERSION: &str = "sdk-version";
+const DAML_SDK_VERSION_ENV: &str = "DAML_SDK_VERSION";
+const DAML_YAML: &str = "daml.yaml";
+
+/// Creates a module with the current Daml version
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("daml");
+    let config: DamlConfig = DamlConfig::try_load(module.config);
+
+    let is_daml_project = context.try_begin_scan()?.set_files(&[DAML_YAML]).is_match();
+
+    if !is_daml_project {
+        return None;
+    }
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_meta(|var, _| match var {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "version" => {
+                    let daml_sdk_version = get_daml_sdk_version(context)?;
+                    VersionFormatter::format_module_version(
+                        module.get_name(),
+                        &daml_sdk_version,
+                        config.version_format,
+                    )
+                    .map(Ok)
+                }
+                _ => None,
+            })
+            .parse(None, Some(context))
+    });
+
+    module.set_segments(match parsed {
+        Ok(segments) => segments,
+        Err(error) => {
+            log::warn!("Error in module `daml`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+fn get_daml_sdk_version(context: &Context) -> Option<String> {
+    context
+        .get_env(DAML_SDK_VERSION_ENV)
+        .or_else(|| read_sdk_version_from_daml_yaml(context))
+}
+
+fn read_sdk_version_from_daml_yaml(context: &Context) -> Option<String> {
+    let file_contents = context.read_file_from_pwd(DAML_YAML)?;
+    let daml_yaml = yaml_rust::YamlLoader::load_from_str(&file_contents).ok()?;
+    let sdk_version = daml_yaml.first()?[DAML_SDK_VERSION].as_str()?;
```

Comment:
It looks like it's safe to do with this type, but please use `get` as `Index` may panic. Perhaps also add a test covering missing `DAML_SDK_VERSION`.

Reply:
I'm not a Rust expert, so I may need some help with this. I had a look at the other uses of this library in this codebase and it looks like the `Index` implementation always returns `Yaml` and that absence is dealt with the `.as_...` methods. Do I understand correctly? Let me write a test as you asked to validate my hypothesis.

Reply:
Test added in https://github.com/starship/starship/pull/4004/commits/8f6e77d998bbb793d4ee4a1081eabc1a69cae98c. IIUC this means that `Index` is safe to use here, right?

Reply:
Just being cautious, as some implementation of `Index` will panic on undefined values.

Reply:
Makes a lot of sense. Can I consider this and https://github.com/starship/starship/pull/4004#discussion_r878697031 resolved?

Reply:
sure.